src/jit/gentree.cpp

```diff
@@ -2516,6 +2573,21 @@ unsigned Compiler::gtHashValue(GenTree* tree)
                     hash += (tree->gtAddrMode.gtOffset << 3) + tree->gtAddrMode.gtScale;
                     break;
 
+                case GT_BLK:
+                case GT_STORE_BLK:
+                    hash += tree->gtBlk.gtBlkSize;
+                    break;
+
+                case GT_OBJ:
+                case GT_STORE_OBJ:
+                    hash ^= reinterpret_cast<unsigned>(tree->AsObj()->gtClass);
```

Comment:
This is the same as the previous `GT_OBJ` handling - though your comment made me realize that I'd failed to remove `GT_OBJ` from the unary case above. I'll remove that.

---
In reply to: [77258271](https://github.com/dotnet/coreclr/pull/7027#discussion_r77258271) [](ancestors = 77258271)